### PR TITLE
Remove wal_receiver mgmt API endpoint

### PR DESCRIPTION
**Pull request by hlinnaka**
Move all the fields that were returned by the wal_receiver endpoint into
timeline_detail. Internally, move those fields from the separate global
WAL_RECEIVERS hash into the LayeredTimeline struct. That way, all the
information about a timeline is kept in one place.

### CODE REVIEW

File: pageserver/src/http/models.rs [hunk 1]
```diff
@@ -140,6 +128,12 @@ pub struct LocalTimelineInfo {
     pub current_logical_size: Option<usize>, // is None when timeline is Unloaded
     pub current_logical_size_non_incremental: Option<usize>,
     pub timeline_state: LocalTimelineState,
+
+    pub wal_source_connstr: Option<String>,
+    #[serde_as(as = "Option<DisplayFromStr>")]
+    pub last_received_msg_lsn: Option<Lsn>,
+    /// the timestamp (in microseconds) of the last received message
```
Comment (SomeoneToIgnore) [inline pageserver/src/http/models.rs:135]: Not related to your PR and no need to change now, but while we at it, let's agree whether this HTTP timeline info API is a human-readable one or not. 

We do not use walreceiver info anywhere so far, AFAIK, so we can change it fluently still.

File: pageserver/src/http/openapi_spec.yml [hunk 2]
```diff
@@ -197,54 +197,6 @@ paths:
               schema:
                 $ref: "#/components/schemas/Error"
 
-  /v1/tenant/{tenant_id}/timeline/{timeline_id}/wal_receiver:
```
Comment (SomeoneToIgnore) [inline pageserver/src/http/openapi_spec.yml:200]: We would need to enumerate new fields in the `TimelineInfo` schema in this file.
Also we should maintain it up to date with https://github.com/neondatabase/cloud/blob/main/swagger/pageserver.yml but for this case it's not super important, I guess.

File: pageserver/src/timelines.rs [hunk 3]
```diff
@@ -37,6 +37,20 @@ pub fn get_info_from_loaded_timeline(
     include_non_incremental_logical_size: bool,
 ) -> anyhow::Result<LocalTimelineInfo> {
     let last_record_lsn = timeline.get_last_record_lsn();
+
+    let (wal_source_connstr, last_received_msg_lsn, last_received_msg_ts) = {
+        let guard = timeline.last_received_wal.lock().unwrap();
+        if let Some(info) = guard.as_ref() {
```
Comment (SomeoneToIgnore) [inline pageserver/src/timelines.rs:43]: Looks like we could leave all fields in our HTTP object without the `Option` wrapper and wrap the object itself in the timeline, as this `if let` logic implies.

**Review by SomeoneToIgnore — APPROVED** [review]
A few nits spotted, nothing big.

File: pageserver/src/http/openapi_spec.yml (hunk 2 above)
Comment (SomeoneToIgnore) [reply pageserver/src/http/openapi_spec.yml:210]: Go code generates API clients, if the methods from the clients were used, those could be affected.
In the end, it's extra fields added to JSON, so missing this update should not disrupt its work on the protocol level.

File: pageserver/src/timelines.rs (hunk 3 above)
Comment (hlinnaka) [reply pageserver/src/timelines.rs:44]: You mean, include the WalReceiverInfo struct itself as a field in LocalTimeLine, like this?
```
diff --git a/pageserver/src/http/models.rs b/pageserver/src/http/models.rs
index dea767ec..669ce724 100644
*** a/pageserver/src/http/models.rs
--- b/pageserver/src/http/models.rs
@@ -131,11 +131,17 @@
     pub current_physical_size_non_incremental: Option<u64>,
     pub timeline_state: LocalTimelineState,
 
-    pub wal_source_connstr: Option<String>,
+    pub wal_receiver_info: Option<WalReceiverInfo>,
+}
+
+#[serde_as]
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct WalReceiverInfo {
+    pub wal_source_connstr: String,
     #[serde_as(as = "Option<DisplayFromStr>")]
-    pub last_received_msg_lsn: Option<Lsn>,
+    pub last_received_msg_lsn: Lsn,
     /// the timestamp (in microseconds) of the last received message
-    pub last_received_msg_ts: Option<u128>,
+    pub last_received_msg_ts: u128,
 }
 
 #[serde_as]
```
Yeah, that would work. A flat response object somehow feels nicer to me, though. I can't really articulate why.
,

Comment (SomeoneToIgnore) [reply pageserver/src/timelines.rs:44]: Yes, but no objections against other variant, since there's no API user that needs this info now, anything works.

**Review by aome510 — APPROVED** [review]

**Pull request closed without merging**